pkg/controller/service/service_controller.go

```diff
@@ -203,13 +207,15 @@ func (s *ServiceController) worker() {
 }
 
 func (s *ServiceController) processNextWorkItem() bool {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I am skimming this PR-- this looks questionable, should it be a TODO context, or re-use the one for the overall controller? Or maybe I misunderstand how contexts are supposed to be used?

Reply:
Controllers are an interesting variant. They are not requests from the outside, so at some point we need to create a root context. Given that I am creating a cancellable context, I did not want it spanning work items. I could create a background context at the worker level, pass it down and then make cancellable contexts from it at the processNextWorkItem() level.